components/sup/doc/http_gateway_butterfly_schema.json

```diff
@@ -0,0 +1,483 @@
+{
+  "definitions": {},
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "description": "Schema for all the data the Habitat supervisor makes available at the /butterfly HTTP API endpoint",
+  "required": [
+    "departure",
+    "election",
+    "election_update",
+    "member",
+    "service",
+    "service_config",
+    "service_file"
+  ],
+  "properties": {
+    "departure": {
+      "type": "object",
+      "description": "A list of departure rumors",
+      "required": [
+        "list",
+        "update_counter"
+      ],
+      "properties": {
+        "list": {
+          "type": "object",
+          "description": "The departure list",
+          "required": [
+            "departure"
+          ],
+          "properties": {
+            "departure": {
+              "type": "object",
+              "description": "Yet another list of departure rumors",
+              "additionalProperties": {
+                "type": "object",
+                "properties": {
+                  "member_id": {
+                    "type": "string",
+                    "description": "The member ID of the member that departed"
+                  }
+                }
+              }
+            }
+          }
+        },
+        "update_counter": {
+          "type": "integer",
+          "description": "A count of how many times the departure rumors have been updated",
+        }
+      }
+    },
+    "election": {
+      "type": "object",
+      "description": "A list of election rumors",
+      "required": [
+        "list",
+        "update_counter"
+      ],
+      "properties": {
+        "list": {
+          "type": "object",
+          "description": "The election list",
+          "additionalProperties": {
+            "type": "object",
+            "description": "The name of a service group",
+            "properties": {
+              "election": {
+                "type": "object",
+                "description": "The election",
+                "properties": {
+                  "from_id": {
+                    "type": "string",
+                    "description": "The member ID being voted for"
+                  },
+                  "member_id": {
+                    "type": "string",
+                    "description": "The member ID being voted for"
+                  },
+                  "service_group": {
+                    "type": "string",
+                    "description": "The service group this election is running in"
+                  },
+                  "status": {
+                    "type": "string",
+                    "description": "The status of the election"
```

Comment:
We should enumerate the possible values here. The more information we can provide users, the better, particularly since we can also automatically generate user-facing documentation from this schema.

Reply:
Will do